x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/SpanFlyout/TruncateHeightSection.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiLink } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { Fragment, useEffect, useRef, useState } from 'react';
+import styled from 'styled-components';
+import { px, units } from '../../../../../../../style/variables';
+// @ts-ignore
+import { Ellipsis } from '../../../../../../shared/Icons';
+
+const ToggleButtonContainer = styled.div`
+  margin-top: ${px(units.half)};
+  user-select: none;
+`;
+
+interface Props {
+  previewHeight: string;
+}
+
+export const TruncateHeightSection: React.SFC<Props> = ({
+  children,
+  previewHeight
+}) => {
+  const [isTruncated, setIsTruncated] = useState(true);
+  const [isContentClipping, setIsContentClipping] = useState(true);
+  const [forcedStaticHeight, setForcedStaticHeight] = useState(true);
+  const contentContainerEl = useRef<HTMLDivElement>(null);
+
+  useEffect(
+    () => {
+      setForcedStaticHeight(true);
+    },
+    [children, previewHeight]
+  );
+
+  useEffect(
+    () => {
+      if (forcedStaticHeight) {
+        if (contentContainerEl.current) {
+          setIsContentClipping(
+            contentContainerEl.current.scrollHeight >
+              contentContainerEl.current.clientHeight
+          );
+        }
+        setForcedStaticHeight(false);
+      }
+    },
+    [forcedStaticHeight]
+  );
```

Comment:
Thanks for trying this out :)
I played a little around, and this seems to do the same:
```js
  const [showToggle, setShowToggle] = useState(true);
  const [isOpen, setIsOpen] = useState(false);
  const ref = useRef(null);

  useEffect(
    () => {
      const shouldShow = ref.current.scrollHeight > previewHeight;
      setShowToggle(shouldShow);
    },
    [children, previewHeight]
  );
```

The trick is to determine whether to show the toggle, if the scroll height is greater than the preview height. Regardless if the content is currently being truncated or not.